flask_resources/serializers/base.py

```diff
@@ -46,29 +47,29 @@ def __init__(
         object_schema_cls,
         list_schema_cls=None,
         schema_context=None,
+        schema_kwargs=None,
         **serializer_options
     ):
         """Initialize the serializer."""
         self.schema_context = schema_context or {}
-        self.object_schema_cls = object_schema_cls
+        schema_kwargs = schema_kwargs or {}
+        self.object_schema = object_schema_cls(context=schema_context, **schema_kwargs)
```

Comment:
minor: now that I look at this, shouldn't we also have below:

```diff
-self.format_serializer_cls = format_serializer_cls(**serializer_options)
+self.format_serializer = format_serializer_cls(**serializer_options)
```

Either that, or we apply the same principle, and store just the classes in `self.format_serializer_cls` and `self.object_schema_cls` and kwargs separately so that they're initialized and used in the methods below... In any case, I would aim for consistency :)

Reply:
Good catch, the `self.format_serializer_cls` is actually not a class but an object :eyes: 

Personally, I'd also go for initializing objects in the constructor, rather than cobbling them together every time we call a method like `dumps()`.
Might just be me, but I find myself trying to figure out the call stacks of certain object methods via `invenio shell` rather often, and then the re-construction of objects in every method is just noise to me.
If you wanna fetch the classes for debugging purposes, `type(self.format_serializer)` or `self.format_serializer.__class__` should still work fine.

Reply:
Personally I also preffer to initialize it, if possible, in the constructor. Or at least have consistency in general.

Reply:
The people have spoken:

- I think we go with initializing in the constructor
- name the class attributes correctly, i.e. without the `_cls` suffix
  - for this, we need to check if it's a breaking change actually for the downstream serializers... if it is then I would open a separate issue and tackle separately